torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1647,9 +1653,18 @@ void ProcessGroupNCCL::watchdogHandler() {
         }
       }
 
+      // a work could be started but not completed, so we should not update
+      // lastStartedSeq_ and lastStartedOpName_ if the work state is checked
+      // multiple times after the start
+      if (lastStartedSeq_ < work.seq_ && work.isStarted()) {
+        lastStartedSeq_ = work.seq_;
+        lastStartedWorkName_ = opTypeToString(work.opType_);
+      }
+
```

Comment:
n00b question: will `work.seq_` always be checked in the numeric order? i.e. the collective work with `seq_=5` won't be checked until `seq_=4` is checked.